app/models/optin/OptInContextData.scala

```diff
@@ -16,16 +16,23 @@
 
 package models.optin
 
+import models.incomeSourceDetails.TaxYear
 import models.itsaStatus.ITSAStatus
 import models.itsaStatus.ITSAStatus.ITSAStatus
 import play.api.libs.json.{Json, OFormat}
 
-case class OptInContextData(crystallisationStatus: Boolean,
-                             previousYearITSAStatus: String,
-                             currentYearITSAStatus: String,
-                             nextYearITSAStatus: String)
+case class OptInContextData(currentTaxYear: String, currentYearITSAStatus: String, nextTaxYear: String, nextYearITSAStatus: String) {
+
+  def currentYearAsTaxYear(): Option[TaxYear] = asTaxYear(currentTaxYear)
+  def nextTaxYearAsTaxYear(): Option[TaxYear] = asTaxYear(nextTaxYear)
```

Comment:
I think storing the tax year in the journey repo as part of the context is a good idea. It's one of the loose ends from Opt Out that is recorded in [MISUV-8167](https://jira.tools.tax.service.gov.uk/browse/MISUV-8167) and I've even started some work on it on a branch that should be visible. In my version, I decided not to to store both tax years, as we can always generate the `nextTaxYear` from the `currentTaxYear`. Do you think that might be worth considering here?

Reply:
if the next-year is always the next consecutive year then yes can change it that way, but I wasn't sure that's the case

Reply:
done